core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -377,5 +414,57 @@ public String location() {
     public void applyTo(TableMetadata.Builder metadataBuilder) {
       metadataBuilder.setLocation(location);
     }
+
+    @Override
+    public void applyTo(ImmutableViewMetadata.Builder viewMetadataBuilder) {
+      viewMetadataBuilder.location(location);
+      viewMetadataBuilder.addChanges(new MetadataUpdate.SetLocation(location));
+    }
+  }
+
+  class AddViewVersion implements MetadataUpdate {
+    private final ViewVersion viewVersion;
+
+    public AddViewVersion(ViewVersion viewVersion) {
+      this.viewVersion = viewVersion;
+    }
+
+    public ViewVersion viewVersion() {
+      return viewVersion;
+    }
+
+    @Override
+    public void applyTo(TableMetadata.Builder metadataBuilder) {
+      // noop
```

Comment:
I think this should throw `UnsupportedOperationException` rather than silently ignoring the problem. If you get here, something went wrong.

Reply:
updated